engine/_classes/class.bertatemplate.php

```diff
@@ -139,17 +140,29 @@ public function addContent($requestURI, $sectionName, &$sections, $tagName, &$ta
         $this->tagName = $tagName;
         $this->tags = &$tags;
 
+        $isEditMode = $this->environment == 'engine';
+
         // add entries...
         $this->content = &$content;
         $this->allContent = &$allContent;
 
+
         $sectionEntriesDS = new SectionEntriesDataService(self::$options['MULTISITE'], $this->sectionName);
-        $entries = $sectionEntriesDS->getByTag($this->tagName, $this->environment == 'engine');
+        $entries = $sectionEntriesDS->getByTag($this->tagName, $isEditMode);
         $siteSectionsDS = new SiteSectionsDataService(self::$options['MULTISITE']);
         $sectionData = $siteSectionsDS->get($this->sectionName);
         $siteSettingsDS = new SiteSettingsDataService(self::$options['MULTISITE']);
         $siteTemplateSettingsDS = new SiteTemplateSettingsDataService(self::$options['MULTISITE'], $this->name);
 
+        $multisitesMenuRenderService = new MultisitesMenuRenderService(
+            self::$options['MULTISITE'],
+            $isEditMode,
+            $siteTemplateSettingsDS->getState()
+        );
```

Comment:
Woud be a good idea to pass `SiteSettingsDataService` in the constructor as it's done below in `SectionEntryRenderService`. Otherwise we are creating to much instances and getting the same data twice. Create a variable `$siteSettings = $siteSettingsDS->getState()` in pass it both constructors.